tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -198,14 +198,12 @@ ZTEST(spi_loopback, test_spi_null_tx_buf)
 {
 	struct spi_dt_spec *spec = loopback_specs[spec_idx];
 	static const uint8_t expected_nop_return_buf[BUF_SIZE] = { 0 };
-	const struct spi_buf_set tx = spi_loopback_setup_xfer(tx_bufs_pool, 1,
-							      NULL, BUF_SIZE);
 	const struct spi_buf_set rx = spi_loopback_setup_xfer(rx_bufs_pool, 1,
 							      buffer_rx, BUF_SIZE);
 
 	(void)memset(buffer_rx, 0x77, BUF_SIZE);
 
-	spi_loopback_transceive(spec, &tx, &rx);
+	spi_loopback_transceive(spec, NULL, &rx);
```

Comment:
Your change is not testing the tx buf to NULL but the tx_buf_set to NULL here (like spi_read() does). I don't see any issue with this test actually. 
You can set the buf to NULL in a buf set and still set a length to n, meaning the driver will have to tx or rx n dummy (or NOP) bytes. It is better to just set the buf set pointer to NULL, but it's not wrong either.

Reply:
Btw, I checked the source of the test prior @decsny great cleanup, and there was a test called spi_null_tx_buf() which was doing exactly that.

Reply:
> Btw, I checked the source of the test prior @decsny great cleanup, and there was a test called spi_null_tx_buf() which was doing exactly that.

should be the same test that is being changed here

Reply:
@tbursztyka the API doc for transceive does say it can be NULL if no tx, I don't think there is an issue with this test but the change should also not be an issue, AFAIK